src/m365/planner/commands/plan/plan-details-get.ts

```diff
@@ -0,0 +1,177 @@
+      this
+        .getPlanDetails(args)
+        .then((res: any): void => {
+          logger.log(res);
+          cb();
+        }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+    }
+    else {
+      this
+        .getGroupId(args)
+        .then((groupId: string): Promise<string> => {
+          this.groupId = groupId;
+          return this.getPlanId(args);
+        })
+        .then((planId: string): Promise<any> => {
+          args.options.planId = planId;
+          return this.getPlanDetails(args);
+        })
+        .then((res: any): void => {
+          logger.log(res);
+          cb();
+        }, (err: any) => this.handleRejectedODataJsonPromise(err, logger, cb));
+    }
+  }
+
+  private getGroupId(args: CommandArgs): Promise<string> {
+    if (args.options.ownerGroupId) {
+      return Promise.resolve(args.options.ownerGroupId);
+    }
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/groups?$filter=displayName eq '${encodeURIComponent(args.options.ownerGroupName as string)}'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: [{ id: string, resourceProvisioningOptions: string[] }] }>(requestOptions)
+      .then(response => {
+        const groupItem: { id: string, resourceProvisioningOptions: string[] } | undefined = response.value[0];
+
+        if (!groupItem) {
+          return Promise.reject(`The specified ownerGroup does not exist`);
+        }
+
+        if (response.value.length > 1) {
+          return Promise.reject(`Multiple ownerGroups with name ${args.options.ownerGroupName} found: Please choose between the following IDs ${response.value.map(x => x.id)}`);
+        }
+
+        return Promise.resolve(groupItem.id);
+      });
+  }
+
+  private async getPlanId(args: CommandArgs): Promise<string> {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/groups/${this.groupId}/planner/plans`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+    
+    return request
+      .get<{ value: any [] }>(requestOptions)
+      .then(response => {
+        const filteredPlan = response.value.filter((plan: any) => plan.title === args.options.planTitle);
+        if (filteredPlan && filteredPlan.length > 0) {
+          if (filteredPlan.length > 1) {
+            return Promise.reject(`Multiple plans with name ${args.options.planTitle} found: ${filteredPlan.map(x => x.id)}`);
+          }
+          return Promise.resolve(filteredPlan[0].id);
+        }
+        return Promise.reject(`The specified plan title does not exist`);
+      });    
+  }
+
+  private getPlanDetails(args: CommandArgs): Promise<any> {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/plans/${args.options.planId}/details`,
+      headers: {
+        'accept': 'application/json'
```

Comment:
Let's use the `application/json;odata.metadata=none` header to avoid retrieving metadata that we don't use